packages/dev/loaders/src/glTF/2.0/Extensions/KHR_animation_pointer.ts

```diff
@@ -0,0 +1,343 @@
+        return this._loader.isExtensionUsed(NAME);
+    }
+
+    /** @hidden */
+    public dispose() {
+        (this._loader as any) = null;
+    }
+
+    /**
+     * according to specification,
+     * It is not allowed to animate a glTFid property, as it does change the structure of the glTF in general
+     * It is not allowed to animate a name property in general.
+     * @param property
+     * @hidden
+     */
+    public accept(property: string): boolean {
+        return property != "name";
+    }
+
+    public loadAnimationAsync(context: string, animation: IAnimation): Nullable<Promise<AnimationGroup>> {
+        // ensure an animation group is present.
+        if (!animation._babylonAnimationGroup) {
+            this._loader.babylonScene._blockEntityCollection = !!this._loader._assetContainer;
+            const group = new AnimationGroup(animation.name || `animation${animation.index}`, this._loader.babylonScene);
+            group._parentContainer = this._loader._assetContainer;
+            this._loader.babylonScene._blockEntityCollection = false;
+            animation._babylonAnimationGroup = group;
+        }
+        const babylonAnimationGroup = animation._babylonAnimationGroup;
+
+        const promises = new Array<Promise<any>>();
+        ArrayItem.Assign(animation.channels);
+        ArrayItem.Assign(animation.samplers);
+
+        for (const channel of animation.channels) {
+            promises.push(this._loadAnimationChannelAsync(`${context}/channels/${channel.index}`, context, animation, channel));
+        }
+
+        return Promise.all(promises).then(() => {
+            babylonAnimationGroup.normalize(0);
+            return babylonAnimationGroup;
+        });
+    }
+
+    /**
+     * @hidden Loads a glTF animation channel.
+     * @param context The context when loading the asset
+     * @param animationContext The context of the animation when loading the asset
+     * @param animation The glTF animation property
+     * @param channel The glTF animation channel property
+     * @param animationTargetOverride The babylon animation channel target override property. My be null.
+     * @returns A void promise when the channel load is complete
+     */
+    public _loadAnimationChannelAsync(
+        context: string,
+        animationContext: string,
+        animation: IAnimation,
+        channel: IAnimationChannel,
+        animationTargetOverride: Nullable<IAnimatable> = null
+    ): Promise<void> {
+        if (channel.target.path != AnimationChannelTargetPath.POINTER) {
+            throw new Error(`${context}/target/path: Invalid value (${channel.target.path})`);
+        }
+
+        if (channel.target.node != undefined) {
+            // According to KHR_animation_pointer specification
+            // If this extension is used, the animation.channel.target.node must not be set.
+            // Because the node isn’t defined, the channel is ignored and not animated due to the specification.

```

Comment:
It's specific to this extension when present, the node property must NOT be set. I do not see where it should be in the core spec. Let me know if i'm wrong.

Reply:
The core spec:
https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#animations
> When node isn’t defined, channel SHOULD be ignored.

I think you are trying to say that if the node is defined, then KHR_animation_pointer shouldn't do anything with this channel? The last sentence is maybe backwards? Do you mean `Because the node is defined`?

Reply:
Yes exactly, i already change the comment to `is defined`

Reply:
Ahh, yes, I didn't see that yet.